config-object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedObjectStore.java

```diff
@@ -154,7 +154,7 @@ public Optional<T> getData(RequestContext context, String id) {
     return getObject(context, id).map(ConfigObject::getData);
   }
 
-  private ContextualConfigObject<T> upsertObject(
+  protected ContextualConfigObject<T> upsertObject(
```

Comment:
This was intentionally private since upsert (above the DB level) is a practice we typically want to avoid. Can you ping me the details of your use case?